python/lsst/daf/butler/core/quantum.py

```diff
@@ -129,14 +131,32 @@ def actualInputs(self):
         """
         return self._actualInputs
 
+    @property
+    def outputs(self):
+        """A `dict` of output datasets (to be) generated for this quantum,
+        with the same form as `predictedInputs`.
+
+        Read-only; update via `addOutput()`.
+        """
+        return self._outputs
+
     def addPredictedInput(self, ref):
         """Add an input `DatasetRef` to the `Quantum`.
 
         This does not automatically update a `Registry`; all `predictedInputs`
         must be present before a `Registry.addQuantum()` is called.
         """
-        datasetTypeName = ref.type.name
+        datasetTypeName = ref.datasetType.name
         if datasetTypeName not in self._actualInputs:
             self._predictedInputs[datasetTypeName] = [ref, ]
         else:
             self._predictedInputs[datasetTypeName].append(ref)
+
+    def addOutput(self, ref):
+        """Add an output `DatasetRef` to the `Quantum`.
+
+        This does not automatically update a `Registry`; all `outputs`
+        must be present before a `Registry.addQuantum()` is called.
```

Comment:
Needs `Parameters` section.

Reply:
Regarding this and all other docstring-related comments - I tried to do as minimal changes as I could, only what was necessary to run my dummy SuperTask example. Looking at the units code I'm not sure what Pim's plans for this are. Right now it looks more like a placeholder and I'd like to understand what is going to happen to it before spending more time on it.

Reply:
After last week's hack session I'm a bit sensitized to adding new code that doesn't include correct docstrings...

Reply:
OK, I'll add something to `Quantum` methods, but `units` needs more work, there is more non-documented stuff there than what I added.